feat(guide): social elements of github and oss

lwasser opened the pull request:
this is a rewrite to the social page. The next step will be cross linking content in this page.

sneakers-the-rat commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+
+## Why social etiquette matters in open source  
+
+Contributing to open source isn’t just about code—it’s about **collaborating with people you may never meet in real life**. Unlike traditional workplaces, GitHub interactions happen asynchronously, across time zones, and often with volunteers.  
```
hell ya

lwasser replied:
@sneakers-the-rat yay! yes my goal with these lessons is to not just teach the technical steps, but also to help people learn that stuff that you and I and others learn "the hard way" by doing it over time. That is part of what I think triggers fear and nervousness. By having our community review all of these lessons, I think collectively, we can pull together our knowledge to make these lessons really great.

lwasser replied:
the goal with this page is to pull together all of the tips we provide on the other pages into a single page as a high level summary.

hamogu commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+:::{admonition} For maintainers
+:class: tip
+**Support contributors to feel confident and welcome**  
+Many new contributors struggle with GitHub’s learning curve and imposter syndrome. If you have bandwidth, try to be sensitive to that. If you don't have bandwidth, gently nudge them to a project that might be a better fit for them if you don't.  
```
```suggestion
Many new contributors struggle with GitHub’s learning curve and imposter syndrome. If you have bandwidth, try to be sensitive to that. If you don't have bandwidth, gently nudge them to a project that might be a better fit for them if you don't.  
```

lwasser replied:
I really appreciate the thoughtful input here, @slobentanzer and @hamogu !! thank you both. I'm working on this locally and will reply to comments once I push the changes. (i may have a few questions / thoughts to discuss too!).

@slobentanzer you may also want to peruse our entire lesson set that we've been developing [here for context](https://www.pyopensci.org/lessons/github-git/your-first-contribution.html)

The goal is that we (and others) can use them in sprints as resources for beginners and people new to the space. This lesson sets the stage for the social elements of open source. But also, social cues are added throughout all of the lessons. 

We ran a review of the first few lessons already, but I plan to open a new pr so folks can review the second half of the lessons too. And of course, issues are always welcome if things are missing, off, etc! If people use these during our summer sprints, I hope we will get even more feedback on them.

slobentanzer replied:
Hi @lwasser, thanks, that's great. I was looking for a resource to not have to reinvent the wheel on the GitHub etiquette onboarding for all the scientists that come without much background there. I'll certainly use it and contribute back my conclusions.

For my context: what kind of contributors are you usually dealing with in these sprints? What is their level (BA, MA, PhD, other), what is their connection to science?

hamogu commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+
+## 2️⃣ Start Small and Contribute Sustainably
+
+Contributing should not be overwhelming and should be sustainable. Starting with small contributions will help you build confidence and credibility while keeping contributions manageable
```
```suggestion
Contributing should not be overwhelming and should be sustainable. Starting with small contributions will help you build confidence and credibility while keeping contributions manageable.
```

slobentanzer reviewed (commented):
Generally very good! Main open question for me (see also the Slack post): how do we account for the "science" part, which is a distinct flavour of open source?

Compare also [here](https://github.com/slolab/.github/blob/main/profile/OPEN_SCIENCE_GUIDE.md).

lwasser commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+
+Contributing to open source isn’t just about code—it’s about **collaborating with people you may never meet in real life**. Unlike traditional workplaces, GitHub interactions happen asynchronously, across time zones, and often with volunteers.  
+
+This means **how you communicate is just as important as what you contribute**.  
```
```suggestion
This **how you communicate is just as important as what you contribute**.  
```

you can modify text inline if you want and add it as a comment

lwasser commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+
+
+:::{admonition} For maintainers
+:class: tip
```
or you can just leave a comment. If you leave multiple comments it's nice to do it as a review vs single comments. this just means all of the comments will appear at one time when you are complete with the review! it means fewer notifications to the repo owners and followers :)

lwasser commented:
@all-contributors please add @slobentanzer for code, review

lwasser commented:
@all-contributors please add @hamogu for code,review

lwasser commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+
+- **Use supportive language in issues and PRs.**  
+  - Instead of "This is wrong," try "Let's adjust this to match the project style." 
+
```
https://output.circle-artifacts.com/output/job/9e995aaa-fe07-4654-8e9f-8ff53b2da6ff/artifacts/0/html/github-git/social-open-source.html#for-maintainers-support-document-and-acknowledge-contributors

Thank you for this!! I decided to revamp the entire maintainer's section (well, make it a section. Check out the preview above). I added all contribs and tried to make that point about acknowledging people's time clear. let me know what you think. 

The other tool I personally LOVE that some don't is pre-commit.ci bot to take that linting setup out of the contributor's hands. They can lint if they want but you can call the bot to lint, too. I really love using it and have found it works well, but some people don't like pre-commit at all (you don't have to set it up). I may add it as a note - also NOTING that some don't like it.

lwasser commented:
> here

Thank you for acknowledging us in your guide!! I think we should add a section on getting started and highlight sprints as a powerful way to do that. We run them yearly, and all the big meetings run them too . They’ve been key in onboarding contributors in our community, yet we haven’t mentioned them in these lessons.

In terms of the science flavor, let's talk more about what you see as being specifically different in the scientific ecosystem.
 
It's tricky because core packages like numpy, pandas are used outside of science. Smaller domain-specific packages may have a different type of community (if a community at all!) that supports them. Let's chat more but perhaps outside of this PR!

lwasser commented on github-git/social-open-source.md:
```diff
@@ -0,0 +1,158 @@
+
+- **Follow the project’s workflow and be responsive.**  
+  - Maintain good communication—if a maintainer asks for changes, reply and iterate.  
+  - If you can’t finish something, let them know so someone else can pick it up.   
```
This is hard! what would you suggest? I had a pr to geopandas that was open for a year! it did get merged and it was a big pr (a full module). We've also had people from sprints work on things for months. 

 I do love the idea of being a bit more specific. I just am not sure how to best word this. But sometimes people just are burnt out and can't work on something they said they would. rather than going dark, i'd prefer they just say - hey - i can't work on this anymore so someone else can pick it up.

slobentanzer replied:
I didn't mean the contributor or maintainer should be obliged to complete the work in two weeks, rather we'd advise to give a response to any of the following results in that time frame:
- positive, I'll do it soon
- positive, but no time (indicate window)
- no bandwidth ATM, someone else feel free to pick up

I feel like no communication leaves everyone in a bad place; it should not be too hard to respond to requests or reminders with one of these options (are there more?). Also, I think one or two weeks for this sort of response is reasonable (except, again, if otherwise agreed on by contributor/maintainer).

willingc commented on contribute-open-source/social-open-source.md:
```diff
@@ -0,0 +1,205 @@
+    > “I’d love to take this on! I’ll start by updating the docs and submit a PR soon.”  
+
+- **Follow the project’s workflow and be responsive.**  
+  - Maintain good communication—if a maintainer asks for changes, reply and iterate.  
```
```suggestion
  - Maintain good communication—if a maintainer asks for changes, reply and iterate.
  - Keep the communication focused on the code and not the person. For example, responding "you're wrong"
    is not helpful; instead, calmly explain the technical approach you took and asking for "Thoughts?" or "Additional ideas" will result in improving the project through respectful collaboration.
```

willingc commented on contribute-open-source/social-open-source.md:
```diff
@@ -0,0 +1,205 @@
+Many contributors hesitate to get involved due to **imposter syndrome or fear of making mistakes**.  
+Maintainers and experienced contributors can help by making GitHub feel more accessible.  
+
+If you have the bandwidth, try to be welcoming and supportive. If you don't have bandwidth, it's ok-—gently nudge new contributors toward beginner-friendly projects or helpful resources.  
```
Also, Pro tip: If a new contributor successfully merges a PR, ask if they would be interested in contributing again. It builds confidence and will often result in more contributor engagement.

willingc commented on contribute-open-source/social-open-source.md:
```diff
@@ -0,0 +1,205 @@
+- **Be patient and approachable.**  
+  - GitHub can be intimidating—help create an environment where people feel comfortable asking questions.
+ 
+- **Give constructive, actionable feedback.**  
```
```suggestion
- **Give direct, constructive, actionable feedback about the contribution content.**  
```

willingc commented on contribute-open-source/social-open-source.md:
```diff
@@ -0,0 +1,205 @@
+
+- **Provide a contribution workflow.**  
+  - If your project has a preferred PR process, document it and link to it in your contributing guidelines to streamline collaboration.  
+
```
Try to add a section about what happens **after** a PR is submitted. Establish norms for time to receive feedback and set contributor expectations.

lwasser replied:
I'm hoping that gets covered in the PR page itself. this page is supposed to be a high level overview of the process!! we can always link to the pages that offer more detailed guidance!!

willingc commented on contribute-open-source/social-open-source.md:
```diff
@@ -0,0 +1,205 @@
+
+## Summary: The social side of open source  
+
+Open source is truly more than just about code. It's about building trust and relationships and, for new contributors, great opportunities to learn and build social, technical, and even management skills. 
```
```suggestion
Open source is truly more than just about code. It's about building trust and relationships and, for new contributors, great opportunities to learn and build social, communication, technical, and even management skills. 
```

willingc reviewed (commented):
Overall, I like this section quite a bit. I've added a few suggestions mostly around communication.

lwasser commented:
Ok friends. I *think* i've addressed all of the comments here. but if i missed some we can always go back and add / update as needed. Thank you for all of the contributions to this lesson. merging it now!! 🚀
